docs/proposals/20200730-ConfigV1Alpha2.md

```diff
@@ -0,0 +1,203 @@
+# Proposal: Config v1alpha2
+
+Author(s): Andrew Rynhard
+
+## Abstract
+
+The goal in this proposal is to outline the v1alpha2 config manifests and to detail how multi-doc YAML enables API and config driven management of Talos without introducing accidental complexity.
+
+## Background
+
+The `v1alpha1` config has limitations, and layout flaws that we would like to fix.
+In our experience with `v1alpha1` we found that a single configuration file made it difficult to evlove configuration options without breaking things.
+
+## Proposal
+
+```yaml
+---
+# Controlplane contains configuration options for the Talos and Kubernetes
+# control planes. The absence of this manifest enables automatic retrieval from
+# the Talos control plane.
+#
+# Dynamic: true
+# Required: false
+kind: controlplane
+version: v1alpha1
+spec:
+  talos:
+    ca:
+      key: ""
+---
+# Join contains the parameters required to join the Talos and Kubernetes control
+# planes.
+#
+# Required: true
+# Dynamic: false
+kind: join
+version: v1alpha1
+spec:
+  kubernetes:
+    endpoint: ""
+    token: ""
+    ca:
+      crt: ""
+  talos:
+    token: ""
+    ca: # Client needs this to validate trustd connection.
+      crt: ""
+---
+# Etcd contains configuration options for the etcd service. The presence of
+# this manifest enables the etcd service.
+#
+# Required: true
+# Dynamic: true
+kind: etcd
+version: v1alpha1
+spec:
+  image: ""
+  args: {}
+---
+# Kubelet contains configuration options for the kubelet service. In the absence
+# of this manifest, a node will pull this manifest from the Talos control plane.
+#
+# Required: true
+# Dynamic: true
+kind: kubelet
+version: v1alpha1
+spec:
+  image: ""
+  args: {}
+---
+# Machine contains machine configuration options.
+#
+# Required: true
+# Dynamic: false
+kind: machine
```

Comment:
I feel like we should just commit to multi-doc and break this up into dedicate configs. Every time I come back to this, it still feels like too much. Most of this could be their own config. They may be small to begin with, but if feels like it leaves room for expansion, and that we aren't in this weird halfway commit place to multi-doc. A narrowly scoped config feels better IMHO.

Reply:
I agree with that fully. And if we do that, we don't have to deal with patching when updating configs - we can simply replace those small pieces of the config. 

I think we could go as small as network interface being a config block.